dbt_semantic_interfaces/parsing/where_filter/where_filter_dimension.py

```diff
@@ -0,0 +1,68 @@
+from __future__ import annotations
+
+from typing import List, Optional, Sequence
+
+from typing_extensions import override
+
+from dbt_semantic_interfaces.call_parameter_sets import TimeDimensionCallParameterSet
+from dbt_semantic_interfaces.parsing.where_filter.parameter_set_factory import (
+    ParameterSetFactory,
+)
+from dbt_semantic_interfaces.parsing.where_filter.query_interface import (
+    QueryInterfaceDimension,
+    QueryInterfaceDimensionFactory,
+)
+from dbt_semantic_interfaces.protocols.protocol_hint import ProtocolHint
+from dbt_semantic_interfaces.type_enums.time_granularity import TimeGranularity
+
+
+class WhereFilterDimension(ProtocolHint[QueryInterfaceDimension]):
+    """A dimension that is passed in through the where filter parameter."""
+
+    @override
+    def _implements_protocol(self) -> QueryInterfaceDimension:
+        return self
+
+    def __init__(  # noqa
+        self,
+        name: str,
+        entity_path: Sequence[str],
+        time_dimension_call_parameter_sets: List[TimeDimensionCallParameterSet],
+    ):
+        self.name = name
+        self.entity_path = entity_path
+        self._time_dimension_call_parameter_sets = time_dimension_call_parameter_sets
+        self.time_granularity: Optional[TimeGranularity] = None
+
+    def grain(self, time_granularity: str) -> QueryInterfaceDimension:
+        """The time granularity."""
+        self.time_granularity = TimeGranularity(time_granularity)
+        self._time_dimension_call_parameter_sets.append(
+            ParameterSetFactory.create_time_dimension(self.name, time_granularity, self.entity_path)
+        )
+        return self
+
+    def alias(self, _alias: str) -> QueryInterfaceDimension:
+        """Renaming the column."""
+        raise NotImplementedError
```

Comment:
Generally I would discourage leaf class `NotImplementedError`s in a PR except when either
1. We're implementing a protocol from another library
2. We plan on fast following in another PR to remove the `NotImplementedError`

I think this falls into (1) given that it sounds like these protocols were lifted from MF? Additionally I'm guess we also plan to implement `alias` shortly to be ready for the 1.7 release of core? If that sounds correct, this is good for now 🙂 

I don't believe 1 applies because